keps/sig-storage/20190204-scale-testing.md

```diff
@@ -0,0 +1,245 @@
+---
+title: Volume Scale and Performance Testing Plan
+authors:
+  - "@msau42"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-scalability
+reviewers:
+  - "@pohly"
+  - "@saad-ali"
+  - "@wojtek-t"
+approvers:
+  - "@saad-ali"
+  - "@wojtek-t"
+editor: TBD
+creation-date: 2019-02-04
+last-updated: 2019-02-14
+status: provisional
+see-also:
+- https://github.com/kubernetes/community/blob/master/sig-scalability/slos/pod_startup_latency.md
+- https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/docs/design.md
+replaces:
+superseded-by:
+---
+
+# Volume Scale and Performance Testing Plan
+
+## Table of Contents
+
+* [Summary](#summary)
+* [Motivation](#motivation)
+   * [Goals](#goals)
+   * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+   * [Test Framework](#test-framework)
+   * [Test Rollout](#test-rollout)
+   * [Test Portability](#test-portability)
+   * [Test Cases](#test-cases)
+      * [Pod Startup](#pod-startup)
+   * [WIP Future Test Cases](#wip-future-test-cases)
+      * [Pod Teardown](#pod-teardown)
+      * [PV Binding Tests](#pv-binding-tests)
+      * [PV Provisioning Tests](#pv-provisioning-tests)
+      * [PV Deletion Tests](#pv-deletion-tests)
+* [Graduation Criteria](#graduation-criteria)
+   * [Phase 1](#phase-1)
+   * [Phase 2](#phase-2)
+   * [Phase 3](#phase-3)
+* [Implementation History](#implementation-history)
+
+
+## Summary
+
+This KEP outlines a plan for testing scalability and performance of K8s storage components.
+
+## Motivation
+
+Adding storage scale and performance tests will help:
+* Understand the current scale limits of the Kuberentes storage system.
+* Set expectations (SLOs) for consumers of the Volume API.
+* Determine bottlenecks and influence which need addressing.
+
+### Goals
+
+* Measure the overhead of K8s components owned by sig-storage:
+  * K8s volume controllers
+  * Kubelet volume manager
+  * CSI sidecars
+* Stress various dimensions of volume operations to determine:
+  * Max volumes per pod
+  * Max volumes per node
+  * Max volumes per cluster
+* Test with the following volume types:
+  * EmptyDir
+  * Secret
+  * Configmap
+  * Downward API
+  * CSI mock driver
+  * TODO: Hostpath?
+  * TODO: Local?
+* Provide tests that vendors can easily run against their volume drivers.
+
+### Non-Goals
+
+* Test and measure storage provider’s drivers.
+
+## Proposal
+
+### Test Framework
+
+The tests should be developed and run in sig-scalability’s test
+[Cluster Loader framework](https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/README.md)
+and infrastructure.
+
+The framework already supports measurements for Pod startup latency and can be
+used to determine a [Pod startup SLO with
+volumes](https://github.com/kubernetes/community/pull/3242).
```

Comment:
If any changes will be needed (and I think there might be) - this should be on us. Happy to be your main point-of-contact with that.
@krzysied - FYI

Reply:
You may want to make it explicit too.